textdb/textdb-textql/src/test/java/edu/uci/ics/textdb/textql/languageparser/TextQLParserTest.java

```diff
@@ -0,0 +1,616 @@
+package edu.uci.ics.textdb.textql.languageparser;
+
+import junit.framework.Assert;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.function.Consumer;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import edu.uci.ics.textdb.textql.languageparser.TextQLParser.Statement;
+import edu.uci.ics.textdb.textql.languageparser.TextQLParser.SelectStatement;
+import edu.uci.ics.textdb.textql.languageparser.TextQLParser.CreateViewStatement;
+import edu.uci.ics.textdb.textql.languageparser.TextQLParser.ExtractPredicate;
+import edu.uci.ics.textdb.textql.languageparser.TextQLParser.KeywordExtractPredicate;
+
+public class TextQLParserTest {
+
+
+    @Before
+    public void setUp() {
+    }
+
+    @Test
+    public void testNumberLiteral() throws ParseException {
```

Comment:
Add a high-level comment to explain the purpose of this case and its main logic.